baseline-refaster-rules/src/main/java/com/palantir/baseline/refaster/AssertjCollectionHasSameSizeAs.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * (c) Copyright 2019 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.baseline.refaster;
+
+import com.google.errorprone.refaster.ImportPolicy;
+import com.google.errorprone.refaster.annotation.AfterTemplate;
+import com.google.errorprone.refaster.annotation.BeforeTemplate;
+import com.google.errorprone.refaster.annotation.UseImportPolicy;
+import java.util.List;
+import org.assertj.core.api.IterableAssert;
+
+public final class AssertjCollectionHasSameSizeAs<T,U> {
+
+    @BeforeTemplate
+    IterableAssert<T> before(IterableAssert<T> assertInProgress, List<U> expected) {
```

Comment:
List can be Iterable

Reply:
Iterable doesn't have the .size() method, so I used Collection initially, but I can't even get my test to pass with this :/

Reply:
I think refaster requires an exact match on the assert type returned by AssertThat:

This should work:

```java
@BeforeTemplate
ListAssert<T> before(ListAssert<T> assertInProgress, Collection<U> expected) {
    return assertInProgress.hasSize(expected.size());
}
```

However it won't apply to collections or iterable targerrs

Reply:
Based on #881 it may or may not work in java 11...